signature/README.md

```diff
@@ -0,0 +1,174 @@
+
+    /// Get the resources list. The list contains resources names to be
+    /// got by the kbs, using `get_resource_from_kbs()` in validate.rs
+    fn needed_resources_list_from_kbs(&self) -> Vec<&str>;
+
+    /// Process all the gathered resources from kbs. All the resources
+    /// gathered from kbs due to the return value of `needed_resources_list_from_kbs()`
+    /// will be recorded into a HashMap, mapping `resource name` ->
+    /// `content of Vec<u8>`. The parameter of this function is the
+    /// HashMap.
+    fn process_gathered_resources(&self, resources: HashMap<&str, Vec<u8>>) -> Result<()>;
+
+    /// Judge whether an image is allowed by this SignScheme.
+    fn allows_image(&self, image: &mut Image) -> Result<()>;
+}
+```
+
+For a specific signing scheme in a Policy Requirement,
+these 5 functions are called in the following order.
+
+```plaintext
++------------+                                          +-----------+
+|            |         1.prepare_runtime_dirs()         |           |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |         2.resources_check()              |           |
+|            +----------------------------------------->|           |
+|            |                                          | signature |
+|  image-rs  |    3.needed_resources_list_from_kbs()    |  schemes  |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |         4.get_scheme_resources()         |           |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |        5.process_gathered_resources      |           |
+|            +----------------------------------------->|           |
+|            |                                          |           |
+|            |                                          |           |
+|            |                                          |           |
++------------+                                          +-----------+
+```
+
+* Firstly, `prepare_runtime_dirs()` will prepare the directories that this
+signing scheme uses, such as the configurations and signature
+files storage dirs.
+* Secondly, `resources_check()` function will return `true`, if some
+resources need to be got from the kbs. If the return value of 
+`resources_check()` is `true`, the following two functions 
+`needed_resources_list_from_kbs()` and `process_gathered_resources()` will
+be called. And if not, neither of them will be called beecause no 
+resources need to be got by AA from kbc.
+* If `resources_check()` returns true, `needed_resources_list_from_kbs()`
+will return a `Vec<&str>`, each element of the `Vec` is a 
+Resource Description, then all these resources will be got from
```

Comment:
The Resource description here is just a `&str` in the code. What does it specifically mean? We should describe it with clearer variable types, at least in the document.

Reply:
Sure!